include/smack/SmackRep.h

```diff
@@ -83,7 +89,16 @@ class SmackRep {
   Program& getProgram() { return program; }
 
 private:
-  void addInit(unsigned region, const Expr* addr, const llvm::Constant* val);
+  // 
+  // Pointer size: 32
+  //
+  static const unsigned ptrsize;
+  //
+  // A flag for bit-vector
+  //
+  static bool BIT_VECTOR;
```

Comment:
At first I used SmackOptions::BitVector to do so. Then the idea of a llvm pass determining whether bit-vector extension is needed came up and changed the implementation. The reason is that the switch to bit-vector might not be controlled only by SmackOptions or manually. Using a static variable give us more flexibility to control the switch since the variable could be set up somewhere in the code by a possibly complex boolean expression representing the priorities of the multiple factors that could affect the switch.